src/types/connection/edge.rs

```diff
@@ -10,27 +10,29 @@ use std::borrow::Cow;
 
 /// The edge type output by the data source
 pub struct Edge<C, T, E> {
-    pub cursor: C,
-    pub element: T,
-    pub additional_fields: E,
+    pub(crate) cursor: C,
+    pub(crate) node: T,
+    pub(crate) additional_fields: E,
 }
 
 impl<C, T, E> Edge<C, T, E> {
+    /// Create a new edge, it can have some additional fields.
     pub fn new_with_additional_fields(cursor: C, element: T, additional_fields: E) -> Self {
```

Comment:
maybe also rename the parameter `node` ? I think it makes things more clear to anyone who have already worked with relay conection